pkg/operator/controller/add.go

```diff
@@ -54,15 +56,35 @@ func AddToManager(mgr manager.Manager, cfg *config.OperatorConfiguration) error
 		return err
 	}
 
+	gardenClientMap, err := clientmapbuilder.
+		NewGardenClientMapBuilder().
+		WithRuntimeClient(mgr.GetClient()).
+		WithClientConnectionConfig(&cfg.VirtualClientConnection).
+		Build(mgr.GetLogger())
+	if err != nil {
+		return fmt.Errorf("failed to build garden ClientMap: %w", err)
+	}
+	if err := mgr.Add(gardenClientMap); err != nil {
+		return err
+	}
+
 	if err := (&garden.Reconciler{
 		Config:                *cfg,
 		Identity:              identity,
 		ImageVector:           imageVector,
 		ComponentImageVectors: componentImageVectors,
+		GardenClientMap:       gardenClientMap,
 	}).AddToManager(mgr); err != nil {
 		return fmt.Errorf("failed adding Garden controller: %w", err)
 	}
 
+	if err := (&care.Reconciler{
+		Config:          *cfg,
+		GardenClientMap: gardenClientMap,
+	}).AddToManager(mgr); err != nil {
+		return fmt.Errorf("failed adding Garden-Care controller: %w", err)
+	}
+
```

Comment:
Can you move this logic to `pkg/operator/controller/garden/add.go`?

Reply:
Now `garden-garden` and `garden-care` controllers share one client-map. This would not be possible anymore then, wouldn't it?
In the shoot case the client map is even a parameter of the outer [add.go](https://github.com/gardener/gardener/blob/7333b8f95696d41a87a91c9fef3578ef6b93547d/pkg/gardenlet/controller/add.go#L59) and it is created in [cmd/gardenlet/app/app.go](https://github.com/gardener/gardener/blob/7333b8f95696d41a87a91c9fef3578ef6b93547d/cmd/gardenlet/app/app.go#L369).